--- a/src/common/file_util.cpp	2019-11-11 21:54:44.000000000 +0100
+++ b/src/common/file_util.cpp	2020-01-14 20:52:55.625876429 +0100
@@ -637,9 +637,9 @@
     } else {
         std::string subdirectory;
         if (envvar == "XDG_DATA_HOME")
-            subdirectory = DIR_SEP ".local" DIR_SEP "share";
+            subdirectory = "saves";
         else if (envvar == "XDG_CONFIG_HOME")
-            subdirectory = DIR_SEP ".config";
+            subdirectory = DIR_SEP "configs";
         else if (envvar == "XDG_CACHE_HOME")
             subdirectory = DIR_SEP ".cache";
         else
