diff --git a/package/mpg123/mpg123.mk b/package/mpg123/mpg123.mk
index dd2d39d..839ef4f 100644
--- a/package/mpg123/mpg123.mk
+++ b/package/mpg123/mpg123.mk
@@ -4,10 +4,10 @@
 #
 ################################################################################
 
-MPG123_VERSION = 1.25.10
+MPG123_VERSION = 1.25.13
 MPG123_SOURCE = mpg123-$(MPG123_VERSION).tar.bz2
 MPG123_SITE = http://downloads.sourceforge.net/project/mpg123/mpg123/$(MPG123_VERSION)
-MPG123_CONF_OPTS = --disable-lfs-alias
+#MPG123_CONF_OPTS = --disable-lfs-alias
 MPG123_INSTALL_STAGING = YES
 MPG123_LICENSE = LGPL-2.1
 MPG123_LICENSE_FILES = COPYING
