diff --git a/Makefile b/Makefile
index 80ab1d14..c53ab7e9 100644
--- a/Makefile
+++ b/Makefile
@@ -157,6 +157,13 @@ else ifeq ($(PLATFORM),n2)
 	CPPFLAGS += $(CPPFLAGS64) -DUSE_RENDER_THREAD
 	AARCH64 = 1
 
+# Odroid Go Ultra - N2+ (SDL2 64-bit)
+else ifeq ($(PLATFORM),s922x)
+	CPUFLAGS = -mcpu=cortex-a73.cortex-a53
+	CPPFLAGS += $(CPPFLAGS64) -DUSE_RENDER_THREAD
+	ASCPUFLAGS = -mcpu=cortex-a73
+	AARCH64 = 1
+
 # Raspberry Pi 3 (SDL2 64-bit)
 else ifeq ($(PLATFORM),rpi3-64-sdl2)
 	CPUFLAGS = -mcpu=cortex-a53
@@ -585,10 +592,14 @@ endif
 
 USE_JIT=1
 
+ifeq ($(ASCPUFLAGS),)
+	ASCPUFLAGS = $(CPUFLAGS)
+endif
+
 ifdef AARCH64
 OBJS += src/osdep/aarch64_helper.o
 src/osdep/aarch64_helper.o: src/osdep/aarch64_helper.s
-	$(AS) $(CPUFLAGS) -o src/osdep/aarch64_helper.o -c src/osdep/aarch64_helper.s
+	$(AS) $(ASCPUFLAGS) -o src/osdep/aarch64_helper.o -c src/osdep/aarch64_helper.s
 else ifeq ($(PLATFORM),$(filter $(PLATFORM),rpi1 rpi1-sdl2))
 OBJS += src/osdep/arm_helper.o
 src/osdep/arm_helper.o: src/osdep/arm_helper.s
