diff --git a/src/osdep/amiberry.cpp b/src/osdep/amiberry.cpp
index 5dc6deee..6edb790e 100644
--- a/src/osdep/amiberry.cpp
+++ b/src/osdep/amiberry.cpp
@@ -688,7 +688,7 @@ void get_data_path(char* out, int size)
 void get_saveimage_path(char* out, int size, int dir)
 {
 	strncpy(out, start_path_data, size - 1);
-	strncat(out, "/savestates/", size - 1);
+	strncat(out, "/recalbox/share/saves/amiga/", size - 1);
 }
 
 void get_configuration_path(char* out, int size)
@@ -763,13 +763,13 @@ void get_rp9_path(char* out, int size)
 void get_savestate_path(char* out, int size)
 {
 	strncpy(out, start_path_data, size - 1);
-	strncat(out, "/savestates/", size - 1);
+	strncat(out, "/recalbox/share/saves/amiga/", size - 1);
 }
 
 void get_screenshot_path(char* out, int size)
 {
 	strncpy(out, start_path_data, size - 1);
-	strncat(out, "/screenshots/", size - 1);
+	strncat(out, "/recalbox/share/screenshots/", size - 1);
 }
 
 int target_cfgfile_load(struct uae_prefs* p, const char* filename, int type, int isdefault)
@@ -1169,8 +1169,8 @@ void load_amiberry_settings(void)
 	int i;
 	strncpy(current_dir, start_path_data, MAX_DPATH - 1);
 	snprintf(config_path, MAX_DPATH, "%s/conf/", start_path_data);
-	snprintf(controllers_path, MAX_DPATH, "%s/controllers/", start_path_data);
-	snprintf(retroarch_file, MAX_DPATH, "%s/conf/retroarch.cfg", start_path_data);
+	snprintf(controllers_path, MAX_DPATH, "%s/conf/", start_path_data);
+	snprintf(retroarch_file, MAX_DPATH, "/recalbox/share/system/configs/retroarch/retroarchcustom.cfg", start_path_data);
 	snprintf(logfile_path, MAX_DPATH, "%s/amiberry.log", start_path_data);
 
 #ifdef ANDROID
@@ -1184,9 +1184,9 @@ void load_amiberry_settings(void)
 	else
 		snprintf(rom_path, MAX_DPATH, "%s/kickstarts/", start_path_data);
 #else
-	snprintf(rom_path, MAX_DPATH, "%s/kickstarts/", start_path_data);
+	snprintf(rom_path, MAX_DPATH, "/recalbox/share/bios/", start_path_data);
 #endif
-	snprintf(rp9_path, MAX_DPATH, "%s/rp9/", start_path_data);
+	snprintf(rp9_path, MAX_DPATH, "%s/conf/", start_path_data);
 	snprintf(path, MAX_DPATH, "%s/conf/amiberry.conf", start_path_data);
 	snprintf(floppy_sounds_dir, MAX_DPATH, "%s/data/floppy_sounds/", start_path_data);
 
@@ -1383,13 +1383,13 @@ int main(int argc, char* argv[])
 #ifdef ANDROID
     strncpy(start_path_data, getenv("EXTERNAL_FILES_DIR"), MAX_DPATH - 1);
 #else
-	getcwd(start_path_data, MAX_DPATH);
+  strncpy(start_path_data, "/tmp/amiga/", MAX_DPATH);
 #endif
 	rename_old_adfdir();
 	load_amiberry_settings();
 	rp9_init();
 
-	snprintf(savestate_fname, sizeof savestate_fname, "%s/savestates/default.ads", start_path_data);
+	snprintf(savestate_fname, sizeof savestate_fname, "/recalbox/share/saves/amiga/default.ads", start_path_data);
 	logging_init();
 
 	memset(&action, 0, sizeof action);
