diff --git a/Makefile b/Makefile
index 248e7c9e..80ab1d14 100644
--- a/Makefile
+++ b/Makefile
@@ -77,8 +77,13 @@ endif
 
 # Raspberry Pi 4 CPU flags
 ifneq (,$(findstring rpi4,$(PLATFORM)))
-	CPUFLAGS = -mcpu=cortex-a72 -mfpu=neon-fp-armv8
+    CPUFLAGS = -mcpu=cortex-a72
+# Raspberry Pi 4 32bits CPU flags
+ifeq (,$(findstring 64,$(PLATFORM)))
+	CPUFLAGS += -mfpu=neon-fp-armv8
 endif
+endif
+
 
 # Mac OS X M1 CPU flags
 ifneq (,$(findstring osx-m1,$(PLATFORM)))
