diff --git a/src/osdep/amiberry.cpp b/src/osdep/amiberry.cpp
index f6f7c988..48211de1 100644
--- a/src/osdep/amiberry.cpp
+++ b/src/osdep/amiberry.cpp
@@ -2350,8 +2350,7 @@ void get_data_path(char* out, int size)
 
 void get_saveimage_path(char* out, int size, int dir)
 {
-	fix_trailing(saveimage_dir);
-	strncpy(out, saveimage_dir, size - 1);
+	strncpy(out, "/recalbox/share/saves/amiga/", size - 1);
 }
 
 void get_configuration_path(char* out, int size)
@@ -2458,8 +2457,7 @@ void get_rp9_path(char* out, int size)
 
 void get_savestate_path(char* out, int size)
 {
-	fix_trailing(savestate_dir);
-	strncpy(out, savestate_dir, size - 1);
+	strncpy(out, "/recalbox/share/saves/amiga/", size - 1);
 }
 
 void fetch_ripperpath(TCHAR* out, int size)
@@ -2482,8 +2480,7 @@ void get_nvram_path(TCHAR *out, int size)
 
 void get_screenshot_path(char* out, int size)
 {
-	fix_trailing(screenshot_dir);
-	strncpy(out, screenshot_dir, size - 1);
+	strncpy(out, "/recalbox/share/screenshots/", size - 1);
 }
 
 int target_cfgfile_load(struct uae_prefs* p, const char* filename, int type, int isdefault)
@@ -3234,8 +3231,8 @@ static void init_amiberry_paths(void)
 {
 	strncpy(current_dir, start_path_data, MAX_DPATH - 1);
 	snprintf(config_path, MAX_DPATH, "%s/conf/", start_path_data);
-	snprintf(controllers_path, MAX_DPATH, "%s/controllers/", start_path_data);
-	snprintf(retroarch_file, MAX_DPATH, "%s/conf/retroarch.cfg", start_path_data);
+	snprintf(controllers_path, MAX_DPATH, "%s/conf/", start_path_data);
+	snprintf(retroarch_file, MAX_DPATH, "/recalbox/share/system/configs/retroarch/retroarchcustom.cfg", start_path_data);
 	snprintf(whdboot_path, MAX_DPATH, "%s/whdboot/", start_path_data);
 	snprintf(logfile_path, MAX_DPATH, "%s/amiberry.log", start_path_data);
 
@@ -3250,9 +3247,9 @@ static void init_amiberry_paths(void)
 	else
 		snprintf(rom_path, MAX_DPATH, "%s/kickstarts/", start_path_data);
 #else
-	snprintf(rom_path, MAX_DPATH, "%s/kickstarts/", start_path_data);
+	snprintf(rom_path, MAX_DPATH, "/recalbox/share/bios/", start_path_data);
 #endif
-	snprintf(rp9_path, MAX_DPATH, "%s/rp9/", start_path_data);
+	snprintf(rp9_path, MAX_DPATH, "%s/conf/", start_path_data);
 #ifdef __MACH__
 	// Open amiberry.conf from Application Data directory
 	const std::string macos_home_directory = getenv("HOME");
@@ -3415,9 +3412,8 @@ int main(int argc, char* argv[])
 	}
 	else
 	{
-		getcwd(start_path_data, MAX_DPATH);
+        strncpy(start_path_data, "/tmp/amiga/", MAX_DPATH);
 	}
-
 	rename_old_adfdir();
 	init_amiberry_paths();
 	// Parse command line to get possibly set amiberry_config.
@@ -3438,8 +3434,7 @@ int main(int argc, char* argv[])
 		abort();
 	}
 
-	fix_trailing(savestate_dir);
-	snprintf(savestate_fname, sizeof savestate_fname, "%s/default.ads", savestate_dir);
+	snprintf(savestate_fname, sizeof savestate_fname, "/recalbox/share/saves/amiga/default.ads", savestate_dir);
 	logging_init();
 
 	memset(&action, 0, sizeof action);
