diff --git a/src/vm/scv/vdp.cpp b/src/vm/scv/vdp.cpp
index 8a54093..dc2c6fe 100755
--- a/src/vm/scv/vdp.cpp
+++ b/src/vm/scv/vdp.cpp
@@ -176,13 +176,13 @@ inline void VDP::draw_text(int dx, int dy, uint8_t data, uint8_t tcol, uint8_t b
 	int dx8 = (dx << 3)+2;
 	int dy16 = dy << 4;
 
-	unsigned char* f = font[data];
+	const unsigned char* f = &font[data];
 	unsigned char* p = (unsigned char*)&text[dy16][dx8];
 	unsigned int c = (tcol << 8) + bcol;
 	if (data)
 	  for(int l = 8; --l >= 0; f++, p += 320)
-	{
-	    unsigned int = *f;
+    {
+	    unsigned int pat = *f;
       dest[0] = (unsigned char)(c >> ((pat >> (7 - 3)) & 8));
       dest[1] = (unsigned char)(c >> ((pat >> (6 - 3)) & 8));
       dest[2] = (unsigned char)(c >> ((pat >> (5 - 3)) & 8));
@@ -225,7 +225,6 @@ inline void VDP::draw_block(int dx, int dy, uint8_t data)
 inline void VDP::draw_graph(int dx, int dy, uint8_t data, uint8_t col)
 {
 	int dx8l = dx << 3;
-	int dx8r = dxl + 4;
 	int dy16 = dy << 4;
 
   uint32_t c = col + (col << 8); c = c + (c << 16);
