diff --git a/package/bcm2835/Config.in b/package/bcm2835/Config.in
index 8c77a62a11..97ae7f458a 100644
--- a/package/bcm2835/Config.in
+++ b/package/bcm2835/Config.in
@@ -1,6 +1,6 @@
 config BR2_PACKAGE_BCM2835
 	bool "bcm2835"
-	depends on BR2_arm
+	depends on BR2_arm || BR2_aarch64
 	help
 	  C library for Broadcom BCM 2835 as used in Raspberry Pi.
 	  It provides access to GPIO and other IO functions on the
