--- a/yabause/src/libretro/libretro.c	2022-07-08 00:43:55.000000000 +0200
+++ b/yabause/src/libretro/libretro.c	2022-07-08 00:49:37.000000000 +0200
@@ -1791,7 +1791,7 @@
    if (does_file_exist(stv_bios_path) != 1)
    {
       log_cb(RETRO_LOG_WARN, "%s NOT FOUND\n", stv_bios_path);
-      snprintf(stv_bios_path, sizeof(stv_bios_path), "%s%cstvbios.zip", g_system_dir, slash);
+      snprintf(stv_bios_path, sizeof(stv_bios_path), "%s%c%s%cstvbios.zip", g_system_dir, slash, "saturn", slash);
       if (does_file_exist(stv_bios_path) != 1)
       {
          log_cb(RETRO_LOG_WARN, "%s NOT FOUND\n", stv_bios_path);
@@ -1802,15 +1802,15 @@
    if (does_file_exist(bios_path) != 1)
    {
       log_cb(RETRO_LOG_WARN, "%s NOT FOUND\n", bios_path);
-      snprintf(bios_path, sizeof(bios_path), "%s%csaturn_bios.bin", g_system_dir, slash);
+      snprintf(bios_path, sizeof(bios_path), "%s%c%s%csaturn_bios.bin", g_system_dir, slash, "saturn", slash);
       if (does_file_exist(bios_path) != 1)
       {
          log_cb(RETRO_LOG_WARN, "%s NOT FOUND\n", bios_path);
-         snprintf(bios_path, sizeof(bios_path), "%s%csega_101.bin", g_system_dir, slash);
+         snprintf(bios_path, sizeof(bios_path), "%s%c%s%csega_101.bin", g_system_dir, slash, "saturn", slash);
          if (does_file_exist(bios_path) != 1)
          {
             log_cb(RETRO_LOG_WARN, "%s NOT FOUND\n", bios_path);
-            snprintf(bios_path, sizeof(bios_path), "%s%cmpr-17933.bin", g_system_dir, slash);
+            snprintf(bios_path, sizeof(bios_path), "%s%c%s%cmpr-17933.bin", g_system_dir, slash, "saturn", slash);
             if (does_file_exist(bios_path) != 1)
             {
                log_cb(RETRO_LOG_WARN, "%s NOT FOUND\n", bios_path);
