--- a/Makefile.libretro
+++ b/Makefile.libretro
@@ -83,15 +83,26 @@
 	LDFLAGS += -shared -Wl,--version-script=$(ROOT_DIR)/link.T
 	fpic = -fPIC
 	GLES = 1
-	GL_LIB := -L/opt/vc/lib -lGLESv2
-	INCFLAGS += -I/opt/vc/include
+	ifneq (,$(findstring rpi4,$(platform)))
+		GL_LIB := -lGLESv2
+		CPUFLAGS += -DNO_ASM -DARM_ASM -frename-registers -ftree-vectorize
+		CFLAGS += -marm -mfloat-abi=hard -mcpu=cortex-a72 -mtune=cortex-a72 -mfpu=neon-vfpv4
+		CXXFLAGS += -marm -mfloat-abi=hard -mcpu=cortex-a72 -mtune=cortex-a72 -mfpu=neon-vfpv4
+		HAVE_NEON = 1
+	else
+		GL_LIB := -L/opt/vc/lib -lbrcmGLESv2
+		INCFLAGS += -I/opt/vc/include
+	endif
 	ifneq (,$(findstring rpi2,$(platform)))
-		CPUFLAGS += -D__arm__ -D__NEON_OPT
-		CFLAGS = -mcpu=cortex-a7 -mfloat-abi=hard
-		CXXFLAGS = -mcpu=cortex-a7 -mfloat-abi=hard
-		HAVE_NEON = 0
+		CPUFLAGS += -DARM -D__arm__ -D__NEON_OPT
+		CPUFLAGS += -mcpu=cortex-a7 -mfpu=neon-vfpv4 -mfloat-abi=hard
+		HAVE_NEON = 1
+	else ifneq (,$(findstring rpi3,$(platform)))
+		CPUFLAGS += -DARM -D__arm__ -D__NEON_OPT
+		CPUFLAGS += -mcpu=cortex-a53 -mfpu=neon-fp-armv8 -mfloat-abi=hard
+		HAVE_NEON = 1
 	else
-		CPUFLAGS += -DARMv5_ONLY 
+		CPUFLAGS += -DARMv5_ONLY
 	endif
 
 # ODROIDs
