--- a/deps/libretro-common/include/glsm/glsm.h	2020-07-02 16:48:25.229322342 +0200
+++ b/deps/libretro-common/include/glsm/glsm.h	2020-07-02 16:50:38.740685430 +0200
@@ -32,7 +32,7 @@
 RETRO_BEGIN_DECLS
 
 #ifdef HAVE_OPENGLES2
-#if defined (IOS) || defined(EMSCRIPTEN)
+#if defined (IOS) || defined(EMSCRIPTEN) || !defined(GL_NV_path_rendering)
 typedef GLfloat GLdouble;
 #endif
 typedef GLclampf GLclampd;
