diff --git a/package/luajit/Config.in b/package/luajit/Config.in
index f4d6001586..b4d714fa79 100644
--- a/package/luajit/Config.in
+++ b/package/luajit/Config.in
@@ -19,7 +19,7 @@ config BR2_PACKAGE_LUAJIT
 	# binaries, so check if that option is supported. See
 	# luajit.mk for details.
 	select BR2_PACKAGE_HAS_LUAINTERPRETER
-	select BR2_HOSTARCH_NEEDS_IA32_COMPILER if !BR2_ARCH_IS_64
+	# select BR2_HOSTARCH_NEEDS_IA32_COMPILER if !BR2_ARCH_IS_64
 	help
 	  LuaJIT implements the full set of language features defined
 	  by Lua 5.1. The virtual machine (VM) is API- and
