From 7d0608e13ce9f7f73c42ffd8d965dc2436db32bf Mon Sep 17 00:00:00 2001
From: David Barbion <davidb@230ruedubac.fr>
Date: Thu, 4 Nov 2021 22:04:49 +0100
Subject: [PATCH 1/2] libpicofe-gl-videocore-build

---
 gl_platform.c | 57 ---------------------------------------------------
 1 file changed, 57 deletions(-)

diff --git a/gl_platform.c b/gl_platform.c
index d5f0c20..4f48f65 100644
--- a/frontend/libpicofe/gl_platform.c
+++ b/frontend/libpicofe/gl_platform.c
@@ -14,27 +14,16 @@
  * couldn't they implement EGL properly? D:
  */
 #include <bcm_host.h>
-#include <X11/Xlib.h>
 #include <dlfcn.h>
 
-static Display *x11display;
-static Window x11window;
 static DISPMANX_DISPLAY_HANDLE_T m_dispmanDisplay;
 static EGL_DISPMANX_WINDOW_T m_nativeWindow;
 
 static void *x11lib;
-#define FPTR(f) typeof(f) * p##f
-static FPTR(XGetGeometry);
-static FPTR(XGetWindowAttributes);
-static FPTR(XTranslateCoordinates);
 
 static void get_window_rect(VC_RECT_T *rect)
 {
-	XWindowAttributes xattrs_root;
 	uint32_t disp_w = 0, disp_h = 0;
-	int dx = 0, dy = 0;
-	unsigned int dw = 0, dh = 0, dummy;
-	Window root, dummyw;
 
 	graphics_get_display_size(0, &disp_w, &disp_h);
 	if (disp_w == 0 || disp_h == 0)
@@ -44,29 +33,6 @@ static void get_window_rect(VC_RECT_T *rect)
 	rect->x = rect->y = 0;
 	rect->width = disp_w;
 	rect->height = disp_h;
-
-	if (x11display == NULL || x11window == 0)
-		return; // use fullscreen
-
-	pXGetGeometry(x11display, x11window, &root, &dx, &dy, &dw, &dh,
-		&dummy, &dummy);
-	pXGetWindowAttributes(x11display, root, &xattrs_root);
-
-	if (dw == xattrs_root.width && dh == xattrs_root.height)
-		return; // use fullscreen
-
-	pXTranslateCoordinates(x11display, x11window, root,
-		dx, dy, &dx, &dy, &dummyw);
-
-	// how to deal with that weird centering thing?
-	// this is not quite right..
-	dx += (disp_w - xattrs_root.width) / 2;
-	dy += (disp_h - xattrs_root.height) / 2;
-
-	rect->x = dx;
-	rect->y = dy;
-	rect->width = dw;
-	rect->height = dh;
 }
 
 static void submit_rect(void)
@@ -94,21 +60,6 @@ static void submit_rect(void)
 
 int gl_platform_init(void **display, void **window, int *quirks)
 {
-	x11display = NULL;
-	x11window = 0;
-
-	x11lib = dlopen("libX11.so.6", RTLD_LAZY);
-	if (x11lib != NULL) {
-		pXGetGeometry = dlsym(x11lib, "XGetGeometry");
-		pXGetWindowAttributes = dlsym(x11lib, "XGetWindowAttributes");
-		pXTranslateCoordinates = dlsym(x11lib, "XTranslateCoordinates");
-		if (pXGetGeometry != NULL && pXGetWindowAttributes != NULL
-		    && pXTranslateCoordinates != NULL)
-		{
-			x11display = *display;
-			x11window = (Window)*window;
-		}
-	}
 
 	bcm_host_init();
 	submit_rect();
@@ -124,14 +75,6 @@ void gl_platform_finish(void)
 {
 	vc_dispmanx_display_close(m_dispmanDisplay);
 	bcm_host_deinit();
-
-	if (x11lib) {
-		dlclose(x11lib);
-		x11lib = NULL;
-	}
-
-	x11display = NULL;
-	x11window = 0;
 }
 
 #else
-- 
2.33.1

