From 4022e068673a77f77c361a2ddf56596538ab733c Mon Sep 17 00:00:00 2001
From: David Barbion <davidb@230ruedubac.fr>
Date: Thu, 4 Nov 2021 22:01:14 +0100
Subject: [PATCH 3/4] mimic-retroarch

---
 frontend/plugin_lib.c | 48 +++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 48 insertions(+)

diff --git a/frontend/plugin_lib.c b/frontend/plugin_lib.c
index ab4d415..8be6532 100644
--- a/frontend/plugin_lib.c
+++ b/frontend/plugin_lib.c
@@ -16,6 +16,7 @@
 #include <fcntl.h>
 #include <unistd.h>
 #include <pthread.h>
+#include <SDL_events.h>
 
 #include "libpicofe/fonts.h"
 #include "libpicofe/input.h"
@@ -612,6 +613,53 @@ static void update_input(void)
 	emu_set_action(emu_act);
 
 	in_keystate = actions[IN_BINDTYPE_PLAYER12];
+
+	if (in_keystate & (1u << DKEY_SELECT))
+  {
+    if (in_keystate & (1u << DKEY_START))
+    {
+      printf("Quit requested\n");
+      SDL_Event quit;
+      quit.type = SDL_QUIT;
+      SDL_PushEvent(&quit);
+    }
+    else if (in_keystate & (1u << DKEY_TRIANGLE))
+    {
+      emu_set_action(SACTION_SAVE_STATE);
+    }
+    else if (in_keystate & (1u << DKEY_SQUARE))
+    {
+      emu_set_action(SACTION_LOAD_STATE);
+    }
+    else if (in_keystate & (1u << DKEY_DOWN))
+    {
+      emu_set_action(SACTION_NEXT_SSLOT);
+    }
+    else if (in_keystate & (1u << DKEY_UP))
+    {
+      emu_set_action(SACTION_PREV_SSLOT);
+    }
+    else if (in_keystate & (1u << DKEY_RIGHT))
+    {
+      emu_set_action(SACTION_FAST_FORWARD);
+    }
+    else if (in_keystate & (1u << DKEY_CROSS))
+    {
+      emu_set_action(SACTION_ENTER_MENU);
+    }
+    else if (in_keystate & (1u << DKEY_CIRCLE))
+    {
+      emu_set_action(SACTION_TOGGLE_FPS);
+    }
+    else if (in_keystate & (1u << DKEY_L1))
+    {
+      emu_set_action(SACTION_TOGGLE_FSKIP);
+    }
+    else if (in_keystate & (1u << DKEY_R1))
+    {
+      emu_set_action(SACTION_SCREENSHOT);
+    }
+  }
 }
 #else /* MAEMO */
 extern void update_input(void);
-- 
2.33.1

