--- pcsx_rearmed-b085fc5ae83c7e0f91cb3a8dc00bd14d2b8f9c1c/libpcsxcore/r3000a.c
+++ pcsx_rearmed-b085fc5ae83c7e0f91cb3a8dc00bd14d2b8f9c1c/libpcsxcore/r3000a.c
@@ -27,7 +27,7 @@
 #include "gte.h"
 
 R3000Acpu *psxCpu = NULL;
-psxRegisters psxRegs;
+extern psxRegisters psxRegs;
 
 int psxInit() {
 	SysPrintf(_("Running PCSX Version %s (%s).\n"), PACKAGE_VERSION, __DATE__);
--- pcsx_rearmed-b085fc5ae83c7e0f91cb3a8dc00bd14d2b8f9c1c/libpcsxcore/psxcounters.c
+++ pcsx_rearmed-b085fc5ae83c7e0f91cb3a8dc00bd14d2b8f9c1c/libpcsxcore/psxcounters.c
@@ -68,7 +68,7 @@
 
 /******************************************************************************/
 
-Rcnt rcnts[ CounterQuantity ];
+extern Rcnt rcnts[ CounterQuantity ];
 
 u32 hSyncCount = 0;
 u32 frame_counter = 0;
