--- ppsspp/UI/NativeApp.cpp.orig	2020-05-28 21:33:08.970814779 +0200
+++ ppsspp/UI/NativeApp.cpp	2020-05-28 21:36:46.881163606 +0200
@@ -539,7 +539,7 @@
 
 	// Note that if we don't have storage permission here, loading the config will
 	// fail and it will be set to the default. Later, we load again when we get permission.
-	g_Config.Load();
+	g_Config.Load("/recalbox/share/system/configs/ppsspp/PSP/SYSTEM/ppsspp.ini", "/recalbox/share/system/configs/ppsspp/PSP/SYSTEM/controls.ini");
 #endif
 	LogManager *logman = LogManager::GetInstance();
 
