--- mpv-0.29.1/wscript.orig	2020-07-25 18:12:57.654014965 +0200
+++ mpv-0.29.1/wscript	2020-07-25 18:21:41.894890398 +0200
@@ -766,7 +766,7 @@
                              "-fgnu89-inline"],
                      linkflags="-L/opt/vc/lib",
                      header_name="bcm_host.h",
-                     lib=['mmal_core', 'mmal_util', 'mmal_vc_client', 'bcm_host']),
+                     lib=['mmal_core', 'mmal_util', 'mmal_vc_client', 'bcm_host', 'vchostif']),
             # We still need all OpenGL symbols, because the vo_gpu code is
             # generic and supports anything from GLES2/OpenGL 2.1 to OpenGL 4 core.
             check_cc(lib="EGL", linkflags="-lGLESv2"),
