--- a/Makefile	2021-04-06 20:13:48.237939105 +0200
+++ b/Makefile	2021-04-06 20:16:16.643032960 +0200
@@ -91,6 +91,23 @@
       endif
    endif
 
+else ifeq ($(platform), odroidgo2)
+   TARGET := $(TARGET_NAME)_libretro.so
+   fpic := -fPIC
+   LDFLAGS := -mcpu=cortex-a35 -flto=16 -fuse-linker-plugin -fprofile-use \
+      -fprofile-dir=/tmp -shared -Wl,--version-script=$(CORE_DIR)/libretro/link.T  -Wl,--no-undefined
+   CFLAGS += -march=armv8-a+crc+simd -mtune=cortex-a35 -mcpu=cortex-a35 \
+                  -Ofast -pipe -fno-plt -fdata-sections -ffunction-sections \
+                  -Wl,--gc-sections -fno-stack-protector -fno-ident \
+                  -fomit-frame-pointer -fmerge-all-constants -fno-math-errno \
+                  -Wl,--gc-sections -falign-functions=1 -falign-jumps=1 \
+                  -falign-loops=1 -fno-unwind-tables -fno-asynchronous-unwind-tables \
+                  -fno-unroll-loops -fmerge-all-constants -frename-registers \
+                  -funsafe-math-optimizations -ftree-vectorize -flto=16 -fuse-linker-plugin
+   CXXFLAGS += $(CFLAGS)
+   CPPFLAGS += $(CFLAGS)
+   ASFLAGS += $(CFLAGS)
+
 # (armv8 a35, hard point, neon based) ###
 # PS Classic
 else ifeq ($(platform), classic_armv8_a35)
 
