diff --git a/package/sdl2/sdl2.mk b/package/sdl2/sdl2.mk
index 6ef530fb43..2f9c20a4b7 100644
--- a/package/sdl2/sdl2.mk
+++ b/package/sdl2/sdl2.mk
@@ -17,7 +17,6 @@ SDL2_CONF_OPTS += \
 	--disable-arts \
 	--disable-esd \
 	--disable-dbus \
-	--disable-pulseaudio \
 	--disable-video-wayland
 
 # We are using autotools build system for sdl2, so the sdl2-config.cmake
@@ -154,6 +153,13 @@ else
 SDL2_CONF_OPTS += --disable-alsa
 endif
 
+ifeq ($(BR2_PACKAGE_PULSEAUDIO),y)
+SDL2_DEPENDENCIES += pulseaudio
+SDL2_CONF_OPTS += --enable-pulseaudio
+else
+SDL2_CONF_OPTS += --disable-pulseaudio
+endif
+
 ifeq ($(BR2_PACKAGE_SDL2_KMSDRM),y)
 SDL2_DEPENDENCIES += libdrm
 SDL2_CONF_OPTS += --enable-video-kmsdrm
@@ -161,4 +167,15 @@ else
 SDL2_CONF_OPTS += --disable-video-kmsdrm
 endif
 
+ifeq ($(BR2_PACKAGE_XSERVER_XORG_SERVER),)
+  ifeq ($(BR2_PACKAGE_MESA3D_OPENGL_EGL),y)
+    TARGET_CFLAGS += -DEGL_NO_X11
+  endif
+endif
+
+# odroid go advance
+ifeq ($(BR2_PACKAGE_LIBRGA),y)
+SDL2_DEPENDENCIES += librga
+endif
+
 $(eval $(autotools-package))
