diff --git a/package/sdl2/sdl2.hash b/package/sdl2/sdl2.hash
index 9e968e0af5..2835cd3fb0 100644
--- a/package/sdl2/sdl2.hash
+++ b/package/sdl2/sdl2.hash
@@ -1,4 +1,4 @@
 # Locally calculated after checking http://www.libsdl.org/release/SDL2-2.0.10.tar.gz.sig
-sha256 b4656c13a1f0d0023ae2f4a9cf08ec92fffb464e0f24238337784159b8b91d57  SDL2-2.0.10.tar.gz
+sha256 349268f695c02efbc9b9148a70b85e58cefbbf704abd3e91be654db7f1e2c863  SDL2-2.0.12.tar.gz
 # Locally calculated
 sha256 a7c8d0b3167c502921192585a582eb8477ad9862330f4d8b9e4a8bb4c1a9d29e  COPYING.txt
