From cc2384e130ca4b4abedaf95e352551ecffda732a Mon Sep 17 00:00:00 2001
From: fewtarius <fewtarius@jelos.org>
Date: Mon, 16 May 2022 23:05:13 +0000
Subject: [PATCH 2/6] Remove a few broken device trees

---
 arch/arm/dts/Makefile | 23 -----------------------
 1 file changed, 23 deletions(-)

diff --git a/arch/arm/dts/Makefile b/arch/arm/dts/Makefile
index 594614a..a7c20e1 100644
--- a/arch/arm/dts/Makefile
+++ b/arch/arm/dts/Makefile
@@ -28,29 +28,6 @@ dtb-$(CONFIG_EXYNOS5) += exynos5250-arndale.dtb \
 	exynos5422-odroidxu3.dtb
 dtb-$(CONFIG_EXYNOS7420) += exynos7420-espresso7420.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += \
-	rk3036-sdk.dtb \
-	rk3066a-mk808.dtb \
-	rk3126-evb.dtb \
-	rk3128-evb.dtb \
-	rk3188-radxarock.dtb \
-	rk3288-evb.dtb \
-	rk3288-fennec.dtb \
-	rk3288-firefly.dtb \
-	rk3288-miqi.dtb \
-	rk3288-phycore-rdk.dtb \
-	rk3288-popmetal.dtb \
-	rk3288-rock2-square.dtb \
-	rk3288-tinker.dtb \
-	rk3288-veyron-jerry.dtb \
-	rk3288-veyron-mickey.dtb \
-	rk3288-veyron-minnie.dtb \
-	rk3288-vyasa.dtb \
-	rk3308-evb.dtb \
-	rk3328-evb.dtb \
-	rk3368-lion.dtb \
-	rk3368-sheep.dtb \
-	rk3368-geekbox.dtb \
-	rk3368-px5-evb.dtb \
 	rk3399-evb.dtb \
 	rk3399-firefly.dtb \
 	rk3399-puma-ddr1333.dtb \
-- 
2.38.1

