From 1a3e974370769f9eaa11179dcde9d1f56d88de63 Mon Sep 17 00:00:00 2001
From: David Barbion <davidb@230ruedubac.fr>
Date: Sat, 19 Nov 2022 23:43:45 +0100
Subject: [PATCH 7/7] set part name

---
 include/configs/rockchip-common.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/configs/rockchip-common.h b/include/configs/rockchip-common.h
index 79ee3d7..4a76766 100644
--- a/include/configs/rockchip-common.h
+++ b/include/configs/rockchip-common.h
@@ -92,7 +92,7 @@
 	"uuid_disk=${uuid_gpt_disk};" \
 	"name=uboot,start=8MB,size=4MB,uuid=${uuid_gpt_loader2};" \
 	"name=resource,start=12MB,size=4MB,uuid=${uuid_gpt_resource};" \
-	"name=ANBERNIC,start=16MB,size=3072M,bootable,uuid=${uuid_gpt_boot};\0"
+	"name=RECALBOX,start=16MB,size=3072M,bootable,uuid=${uuid_gpt_boot};\0"
 #else
 #define PARTS_DEFAULT \
 	"uuid_disk=${uuid_gpt_disk};" \
-- 
2.38.1

