--- a/libretro.cpp	2022-07-07 19:56:26.000000000 +0200
+++ b/libretro.cpp	2022-07-07 19:58:38.000000000 +0200
@@ -1218,13 +1218,13 @@
    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
    {
       if (strcmp(var.value, "System Card 3") == 0)
-         setting_pce_fast_cdbios = "syscard3.pce";
+         setting_pce_fast_cdbios = "pcengine/syscard3.pce";
       else if (strcmp(var.value, "System Card 2") == 0)
-         setting_pce_fast_cdbios = "syscard2.pce";
+         setting_pce_fast_cdbios = "pcengine/syscard2.pce";
       else if (strcmp(var.value, "System Card 1") == 0)
-         setting_pce_fast_cdbios = "syscard1.pce";
+         setting_pce_fast_cdbios = "pcengine/syscard1.pce";
       else if (strcmp(var.value, "Games Express") == 0)
-         setting_pce_fast_cdbios = "gexpress.pce";
+         setting_pce_fast_cdbios = "pcengine/gexpress.pce";
       else if (strcmp(var.value, "System Card 3 US") == 0)
          setting_pce_fast_cdbios = "syscard3u.pce";
       else if (strcmp(var.value, "System Card 2 US") == 0)
