diff --git a/package/libwebsockets/libwebsockets.mk b/package/libwebsockets/libwebsockets.mk
index dedd03347d..8e6d02205a 100644
--- a/package/libwebsockets/libwebsockets.mk
+++ b/package/libwebsockets/libwebsockets.mk
@@ -4,13 +4,17 @@
 #
 ################################################################################
 
-LIBWEBSOCKETS_VERSION = 3.2.2
+LIBWEBSOCKETS_VERSION = 4.0.20
 LIBWEBSOCKETS_SITE = $(call github,warmcat,libwebsockets,v$(LIBWEBSOCKETS_VERSION))
-LIBWEBSOCKETS_LICENSE = LGPL-2.1 with exceptions
+LIBWEBSOCKETS_LICENSE = MIT with exceptions
 LIBWEBSOCKETS_LICENSE_FILES = LICENSE
 LIBWEBSOCKETS_DEPENDENCIES = zlib
 LIBWEBSOCKETS_INSTALL_STAGING = YES
-LIBWEBSOCKETS_CONF_OPTS = -DLWS_WITHOUT_TESTAPPS=ON -DLWS_IPV6=ON
+LIBWEBSOCKETS_CONF_OPTS = \
+	-DLWS_WITHOUT_TESTAPPS=ON \
+	-DLWS_IPV6=ON \
+	-DLWS_UNIX_SOCK=ON \
+	-DLWS_WITHOUT_EXTENSIONS=OFF
 
 # If LWS_MAX_SMP=1, then there is no code related to pthreads compiled
 # in the library. If unset, LWS_MAX_SMP defaults to 32 and a small
@@ -44,6 +48,20 @@ LIBWEBSOCKETS_CONF_OPTS += \
 	-DLWS_WITH_LIBEVENT=OFF
 endif
 
+ifeq ($(BR2_PACKAGE_LIBGLIB2),y)
+LIBWEBSOCKETS_DEPENDENCIES += libglib2
+LIBWEBSOCKETS_CONF_OPTS += -DLWS_WITH_GLIB=ON
+else
+LIBWEBSOCKETS_CONF_OPTS += -DLWS_WITH_GLIB=OFF
+endif
+
+ifeq ($(BR2_PACKAGE_UTIL_LINUX_LIBMOUNT),y)
+LIBWEBSOCKETS_DEPENDENCIES += util-linux
+LIBWEBSOCKETS_CONF_OPTS += -DLWS_WITH_FSMOUNT=ON
+else
+LIBWEBSOCKETS_CONF_OPTS += -DLWS_WITH_FSMOUNT=OFF
+endif
+
 ifeq ($(BR2_PACKAGE_LIBUV),y)
 LIBWEBSOCKETS_DEPENDENCIES += libuv
 LIBWEBSOCKETS_CONF_OPTS += -DLWS_WITH_LIBUV=ON
