diff --git a/package/nodejs/nodejs.mk b/package/nodejs/nodejs.mk
index 3f35ac7c33..acbc7a0131 100644
--- a/package/nodejs/nodejs.mk
+++ b/package/nodejs/nodejs.mk
@@ -65,6 +65,8 @@ define HOST_NODEJS_CONFIGURE_CMDS
 		--shared-openssl-includes=$(HOST_DIR)/include/openssl \
 		--shared-openssl-libpath=$(HOST_DIR)/lib \
 		--shared-zlib \
+		--shared-zlib-includes=$(HOST_ZLIB_SRCDIR) \
+		--shared-zlib-libpath=$(HOST_DIR)/lib \
 		--no-cross-compiling \
 		--with-intl=small-icu \
 	)
