--- a/Source/Core/Core/Config/MainSettings.cpp	2022-07-21 20:51:06.545821763 +0200
+++ b/Source/Core/Core/Config/MainSettings.cpp	2022-07-21 20:52:29.840040369 +0200
@@ -576,11 +576,8 @@
 
 std::string GetBootROMPath(const std::string& region_directory)
 {
-  const std::string path =
-      File::GetUserPath(D_GCUSER_IDX) + DIR_SEP + region_directory + DIR_SEP GC_IPL;
-  if (!File::Exists(path))
-    return File::GetSysDirectory() + GC_SYS_DIR + DIR_SEP + region_directory + DIR_SEP GC_IPL;
-  return path;
+  File::GetUserPath(D_GCUSER_IDX); // on garde ça au cas ou..
+  return std::string("/recalbox/share/bios/gamecube") + DIR_SEP + region_directory + DIR_SEP GC_IPL;
 }
 
 std::string GetMemcardPath(ExpansionInterface::Slot slot, DiscIO::Region region, u16 size_mb)
