From 86da87c5b90f703f7b8e6aaac10c5d86f8be3213 Mon Sep 17 00:00:00 2001
From: David Barbion <davidb@230ruedubac.fr>
Date: Tue, 14 Feb 2023 23:27:38 +0100
Subject: [PATCH 2/4] enable hwid dtb for rg353x

---
 scripts/mkmultidtb.py | 21 ++++++++++++++-------
 1 file changed, 14 insertions(+), 7 deletions(-)

diff --git a/scripts/mkmultidtb.py b/scripts/mkmultidtb.py
index 81b814acd..d62706d7f 100755
--- a/scripts/mkmultidtb.py
+++ b/scripts/mkmultidtb.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # SPDX-License-Identifier: (GPL-2.0+ OR MIT)
 # Copyright (c) 2018 Fuzhou Rockchip Electronics Co., Ltd
 #
@@ -20,15 +20,21 @@ from collections import OrderedDict
 DTBS = {}
 
 DTBS['PX30-EVB'] = OrderedDict([('px30-evb-ddr3-v10', '#_saradc_ch0=1024'),
-				('px30-evb-ddr3-lvds-v10', '#_saradc_ch0=512')])
+                                ('px30-evb-ddr3-lvds-v10', '#_saradc_ch0=512')])
 
 DTBS['RK3308-EVB'] = OrderedDict([('rk3308-evb-dmic-i2s-v10', '#_saradc_ch3=288'),
-				  ('rk3308-evb-dmic-pdm-v10', '#_saradc_ch3=1024'),
-				  ('rk3308-evb-amic-v10', '#_saradc_ch3=407')])
+                                  ('rk3308-evb-dmic-pdm-v10', '#_saradc_ch3=1024'),
+                                  ('rk3308-evb-amic-v10', '#_saradc_ch3=407')])
+
+DTBS['rg353x'] = OrderedDict([('rk3566-anbernic-rg353p', '#_saradc_ch1=852'),
+                              ('rk3566-anbernic-rg353v', '#_saradc_ch1=681'),
+                              ('rk3566-anbernic-rg353m', '#_saradc_ch1=512'),
+                              ('rk3566-anbernic-rg503', '#_saradc_ch1=1023')])
+
 
 def main():
     if (len(sys.argv) < 2) or (sys.argv[1] == '-h'):
-        print __doc__
+        print(__doc__)
         sys.exit(2)
 
     BOARD = sys.argv[1]
@@ -47,9 +53,10 @@ def main():
         shutil.copyfile(ori_file, new_file)
         target_dtb_list += ' ' + new_file
 
-    print target_dtb_list
-    os.system('scripts/resource_tool logo.bmp logo_kernel.bmp ' + target_dtb_list)
+    print(target_dtb_list)
+    os.system('scripts/resource_tool logo.bmp logo_kernel.bmp battery_0.bmp battery_1.bmp battery_2.bmp battery_3.bmp battery_4.bmp battery_5.bmp battery_fail.bmp ' + target_dtb_list)
     os.system('rm ' + target_dtb_list)
 
+
 if __name__ == '__main__':
     main()
-- 
2.39.1

