--- bluez-5.54/plugins/sixaxis.c.orig	2020-07-04 16:33:22.098330563 +0200
+++ bluez-5.54/plugins/sixaxis.c	2020-07-04 16:34:28.637752991 +0200
@@ -364,7 +364,11 @@
 	info("sixaxis: setting up new device");
 
 	btd_device_device_set_name(device, cp->name);
-	btd_device_set_pnpid(device, cp->source, cp->vid, cp->pid, cp->version);
+	/* if device reports different pid/vid on BT prefer those over USB */
+	if (cp->bt_pid)
+		btd_device_set_pnpid(device, cp->source, cp->vid, cp->bt_pid, cp->version);
+	else
+		btd_device_set_pnpid(device, cp->source, cp->vid, cp->pid, cp->version);
 	btd_device_set_trusted(device, false);
 	btd_device_set_temporary(device, true);
 
--- bluez-5.54/profiles/input/sixaxis.h.orig	2020-07-04 16:36:39.680609726 +0200
+++ bluez-5.54/profiles/input/sixaxis.h	2020-07-04 16:35:18.717317026 +0200
@@ -38,6 +38,7 @@
 	uint16_t vid;
 	uint16_t pid;
 	uint16_t version;
+  uint16_t bt_pid;
 	CablePairingType type;
 };
 
@@ -59,6 +60,7 @@
 			.vid = 0x054c,
 			.pid = 0x042f,
 			.version = 0x0000,
+      .bt_pid = 0x0268,
 			.type = CABLE_PAIRING_SIXAXIS,
 		},
 		{
