diff --git a/package/Config.in b/package/Config.in
index f7215f45dd..6e704e7bc8 100644
--- a/package/Config.in
+++ b/package/Config.in
@@ -342,7 +342,6 @@ comment "Graphic libraries"
 	source "package/ocrad/Config.in"
 	source "package/ogre/Config.in"
 	source "package/psplash/Config.in"
-	source "package/sdl/Config.in"
 	source "package/sdl_gfx/Config.in"
 	source "package/sdl_image/Config.in"
 	source "package/sdl_mixer/Config.in"
