diff --git a/package/kodi/Config.in b/package/kodi/Config.in
index 2acb271992..c592b30df2 100644
--- a/package/kodi/Config.in
+++ b/package/kodi/Config.in
@@ -17,6 +17,39 @@ comment "kodi needs python w/ .py modules, a uClibc or glibc toolchain w/ C++, t
 config BR2_PACKAGE_KODI_PLATFORM_SUPPORTS
 	bool
 
+config BR2_PACKAGE_KODI_PLATFORM_SUPPORTS_GBM_GL
+	bool
+	default y
+	depends on BR2_ENABLE_LOCALE # libinput
+	depends on BR2_PACKAGE_HAS_LIBGL
+	depends on BR2_PACKAGE_HAS_LIBEGL
+	depends on BR2_PACKAGE_MESA3D_GBM
+	depends on BR2_PACKAGE_HAS_UDEV
+	depends on !BR2_PACKAGE_RPI_USERLAND
+	select BR2_PACKAGE_KODI_PLATFORM_SUPPORTS
+
+config BR2_PACKAGE_KODI_PLATFORM_SUPPORTS_GBM_GLES
+	bool
+	default y
+	depends on BR2_ENABLE_LOCALE # libinput
+	depends on BR2_PACKAGE_HAS_LIBGLES
+	depends on BR2_PACKAGE_HAS_UDEV
+	depends on !BR2_PACKAGE_RPI_USERLAND
+	depends on !BR2_cortex_a72
+	select BR2_PACKAGE_KODI_PLATFORM_SUPPORTS
+
+config BR2_PACKAGE_KODI_PLATFORM_SUPPORTS_RBPI4
+	bool
+	default y
+	# List of valid CPUs can be found here:
+	# https://github.com/xbmc/xbmc/blob/Krypton/project/cmake/scripts/rbpi/ArchSetup.cmake#L12
+	depends on BR2_cortex_a72
+	depends on !BR2_PACKAGE_RPI_USERLAND
+	depends on BR2_PACKAGE_HAS_LIBGLES
+	depends on BR2_PACKAGE_HAS_UDEV # libinput
+	depends on BR2_PACKAGE_MESA3D_GBM
+	select BR2_PACKAGE_KODI_PLATFORM_SUPPORTS
+
 config BR2_PACKAGE_KODI_PLATFORM_SUPPORTS_RBPI
 	bool
 	default y
@@ -116,6 +149,22 @@ if BR2_PACKAGE_KODI
 
 source "package/kodi-platform/Config.in"
 
+choice
+	prompt "Kodi flavor"
+	default BR2_PACKAGE_KODI_FLAVOR_XBMC
+	help
+	  Choose between many flavors
+	  * xbmc: official Kodi release
+	  * popcornmix: Raspberry PI optimized release
+
+config BR2_PACKAGE_KODI_FLAVOR_XBMC
+	bool "xbmc"
+
+config BR2_PACKAGE_KODI_FLAVOR_POPCORNMIX
+	bool "popcornmix"
+
+endchoice
+
 config BR2_PACKAGE_KODI_ALSA_LIB
 	bool "alsa"
 	select BR2_PACKAGE_ALSA_LIB
@@ -198,8 +247,27 @@ choice
 	prompt "platform"
 	default BR2_PACKAGE_KODI_PLATFORM_X11_OPENGL
 
+config BR2_PACKAGE_KODI_PLATFORM_GBM_GL
+	bool "GBM/GL (w/o X11)"
+	depends on BR2_PACKAGE_KODI_PLATFORM_SUPPORTS_GBM_GL
+	select BR2_PACKAGE_LIBGLU
+	select BR2_PACKAGE_LIBINPUT
+	select BR2_PACKAGE_LIBXKBCOMMON
+
+config BR2_PACKAGE_KODI_PLATFORM_GBM_GLES
+	bool "GBM/GLES (w/o X11)"
+	depends on BR2_PACKAGE_KODI_PLATFORM_SUPPORTS_GBM_GLES
+	select BR2_PACKAGE_LIBINPUT
+	select BR2_PACKAGE_LIBXKBCOMMON
+
+config BR2_PACKAGE_KODI_PLATFORM_RBPI4
+	bool "Raspberry Pi4 only"
+	depends on BR2_PACKAGE_KODI_PLATFORM_SUPPORTS_RBPI4
+	select BR2_PACKAGE_LIBINPUT
+	select BR2_PACKAGE_LIBXKBCOMMON
+
 config BR2_PACKAGE_KODI_PLATFORM_RBPI
-	bool "Raspberry Pi"
+	bool "Raspberry Pi (not 4)"
 	depends on BR2_PACKAGE_KODI_PLATFORM_SUPPORTS_RBPI
 	select BR2_PACKAGE_LIBINPUT
 	select BR2_PACKAGE_LIBXKBCOMMON
