--- a/Makefile
+++ b/Makefile
@@ -155,6 +155,17 @@
       endif
    endif
 
+   # ROCKCHIP
+   ifneq (,$(findstring rockchip,$(platform)))
+      GLES = 1
+      FORCE_GLES=1
+      GL_LIB := -lGLESv2
+      CPUFLAGS += -DNO_ASM -DARM -D__arm__ -DARM_ASM -D__NEON_OPT -DNOSSE
+      HAVE_NEON = 1
+      WITH_DYNAREC=arm
+      CPUFLAGS += -marm -mcpu=cortex-a53 -mfpu=neon-fp-armv8 -mfloat-abi=hard
+   endif
+
    # ODROIDs
    ifneq (,$(findstring odroid,$(platform)))
       BOARD := $(shell cat /proc/cpuinfo | grep -i odroid | awk '{print $$3}')
