--- a/libretro/libretro-glue.h	
+++ b/libretro/libretro-glue.h
@@ -166,13 +166,13 @@
 "
 
 // Amiga kickstarts
-#define A500_ROM                "kick34005.A500"
-#define A500KS2_ROM             "kick37175.A500"
-#define A600_ROM                "kick40063.A600"
-#define A1200_ROM               "kick40068.A1200"
-#define A4000_ROM               "kick40068.A4000"
-#define CD32_ROM                "kick40060.CD32"
-#define CD32_ROM_EXT            "kick40060.CD32.ext"
+#define A500_ROM                "kick34005.A500.rom"
+#define A500KS2_ROM             "kick37175.A500.rom"
+#define A600_ROM                "kick40063.A600.rom"
+#define A1200_ROM               "kick40068.A1200.rom"
+#define A4000_ROM               "kick40068.A4000.rom"
+#define CD32_ROM                "kick40060.CD32.rom"
+#define CD32_ROM_EXT            "kick40060.CD32.ext.rom"
 
 // Amiga files
 #define ADF_FILE_EXT            "adf"
--- a/whdload/WHDLoad_files/S/Startup-Sequence
+++ b/whdload/WHDLoad_files/S/Startup-Sequence
@@ -36,21 +36,21 @@
 
 Assign EXISTS RASystem: >NIL:
 IF NOT WARN
-    IF NOT EXISTS WHDLoad:Devs/Kickstarts/kick33180.A500
-       IF EXISTS RASystem:kick33180.A500
-          Copy RASystem:kick33180.A500 WHDLoad:Devs/Kickstarts/
+    IF NOT EXISTS WHDLoad:Devs/Kickstarts/kick33180.A500.rom
+       IF EXISTS RASystem:kick33180.A500.rom
+          Copy RASystem:kick33180.A500.rom WHDLoad:Devs/Kickstarts/
        ENDIF
     ENDIF
 
-    IF NOT EXISTS WHDLoad:Devs/Kickstarts/kick34005.A500
-       IF EXISTS RASystem:kick34005.A500
-          Copy RASystem:kick34005.A500 WHDLoad:Devs/Kickstarts/
+    IF NOT EXISTS WHDLoad:Devs/Kickstarts/kick34005.A500.rom
+       IF EXISTS RASystem:kick34005.A500.rom
+          Copy RASystem:kick34005.A500.rom WHDLoad:Devs/Kickstarts/
        ENDIF
     ENDIF
 
-    IF NOT EXISTS WHDLoad:Devs/Kickstarts/kick40068.A1200
-       IF EXISTS RASystem:kick40068.A1200
-          Copy RASystem:kick40068.A1200 WHDLoad:Devs/Kickstarts/
+    IF NOT EXISTS WHDLoad:Devs/Kickstarts/kick40068.A1200.rom
+       IF EXISTS RASystem:kick40068.A1200.rom
+          Copy RASystem:kick40068.A1200.rom WHDLoad:Devs/Kickstarts/
        ENDIF
     ENDIF
 
