diff --git a/Base/CPU.cpp b/Base/CPU.cpp
index 8855861..8520aca 100644
--- a/Base/CPU.cpp
+++ b/Base/CPU.cpp
@@ -387,9 +387,21 @@ void ExecuteChunk()
 // The main Z80 emulation loop
 void Run()
 {
+    bool NeedCheckAutoRun = true;
+
     // Loop until told to quit
     while (UI::CheckEvents())
     {
+        // Recalbox-Patch - autoboot disk 1
+        if (NeedCheckAutoRun)
+        {
+          if ((GetOption(autoload)) && (IO::IsAtStartupScreen()))
+          {
+              pFloppy1->Insert(GetOption(disk1), true);
+              NeedCheckAutoRun = false;
+          }
+        }
+
         if (g_fPaused)
             continue;
 
