--- a/src/video/ffmpeg.c	2018-01-14 09:09:59.000000000 -0500
+++ b/src/video/ffmpeg.c	2018-05-25 20:43:49.296264232 -0400
@@ -96,7 +96,7 @@
 
   if (perf_lvl & LOW_LATENCY_DECODE)
     // Use low delay single threaded encoding
-    decoder_ctx->flags |= CODEC_FLAG_LOW_DELAY;
+    decoder_ctx->flags |= AV_CODEC_FLAG_LOW_DELAY;
 
   if (perf_lvl & SLICE_THREADING)
     decoder_ctx->thread_type = FF_THREAD_SLICE;
--- a/src/video/sdl.c	2018-01-14 09:09:59.000000000 -0500
+++ b/src/video/sdl.c	2018-05-25 20:46:35.792264232 -0400
@@ -40,7 +40,7 @@
     return -1;
   }
   
-  ffmpeg_buffer = malloc(DECODER_BUFFER_SIZE + FF_INPUT_BUFFER_PADDING_SIZE);
+  ffmpeg_buffer = malloc(DECODER_BUFFER_SIZE + AV_INPUT_BUFFER_PADDING_SIZE);
   if (ffmpeg_buffer == NULL) {
     fprintf(stderr, "Not enough memory\n");
     ffmpeg_destroy();
