--- a/src/libretro.c.org	2022-11-01 19:49:06.952794694 +0100
+++ b/src/libretro.c	2022-11-01 19:49:12.392846899 +0100
@@ -527,7 +527,7 @@
    info->library_version = version;
    info->need_fullpath = false;
    info->block_extract = false;
-   info->valid_extensions = "tzx|tap|z80|rzx|scl|trd|dsk|zip";
+   info->valid_extensions = "csw|ltp|pzx|spc|sta|tap|tzx|wav|slt|sna|mgt|sp|szx|z80|zxs|d40|d80|dsk|img|fdi|mgt|opd|opu|sad|scl|td0|trd|udi|rzx|dck|hdf|mdr";
 }
 
 void retro_set_environment(retro_environment_t cb)
