--- ./gamecon_gpio_rpi-1.4/Makefile.orig	2020-04-23 21:23:55.579545267 +0200
+++ ./gamecon_gpio_rpi-1.4/Makefile	2020-04-23 21:25:45.970498345 +0200
@@ -1,8 +1,23 @@
-obj-m := gamecon_gpio_rpi.o
+ifneq (${KERNELRELEASE},)
+	obj-m := gamecon_gpio_rpi.o
+else
+	KERNELDIR        ?= /lib/modules/$(shell uname -r)/build
+	MODULE_DIR       ?= $(shell pwd)
+	ARCH             ?= $(shell uname -i)
+	CROSS_COMPILE    ?=
+	INSTALL_MOD_PATH ?= /
+endif
 KVERSION := `uname -r`
 
-all:
-	$(MAKE) -C /lib/modules/$(KVERSION)/build M=$(PWD) modules
+all: modules
+
+modules:
+	${MAKE} ARCH="${ARCH}" CROSS_COMPILE="${CROSS_COMPILE}" -C ${KERNELDIR} SUBDIRS="${MODULE_DIR}"  modules
+
+modules_install:
+	${MAKE} ARCH="${ARCH}" CROSS_COMPILE="${CROSS_COMPILE}" INSTALL_MOD_PATH="${INSTALL_MOD_PATH}" -C ${KERNELDIR} SUBDIRS="${MODULE_DIR}"  modules_install
 
 clean:
-	$(MAKE) -C /lib/modules/$(KVERSION)/build M=$(PWD) clean
+	rm -f *.o *.ko *.mod.c .*.o .*.ko .*.mod.c .*.cmd *~
+	rm -f Module.symvers Module.markers modules.order
+	rm -rf .tmp_versions
