diff --git a/pcsx2/CDVD/CDVD.cpp b/pcsx2/CDVD/CDVD.cpp
index 2921b4f8f..0983c112e 100644
--- a/pcsx2/CDVD/CDVD.cpp
+++ b/pcsx2/CDVD/CDVD.cpp
@@ -25,7 +25,7 @@
 #include "CDVD_internal.h"
 #include "CDVDisoReader.h"
 
-#include "GS.h" // for gsVideoMode
+#include "pcsx2/GS.h" // for gsVideoMode
 #include "Elfheader.h"
 #include "ps2/BiosTools.h"
 
diff --git a/pcsx2/DebugTools/Debug.h b/pcsx2/DebugTools/Debug.h
index 87ed02bfa..342113c7c 100644
--- a/pcsx2/DebugTools/Debug.h
+++ b/pcsx2/DebugTools/Debug.h
@@ -16,7 +16,7 @@
 #pragma once
 
 #include "common/TraceLog.h"
-#include "Config.h"
+#include "pcsx2/Config.h"
 #include "Memory.h"
 
 extern FILE *emuLog;
diff --git a/pcsx2/DebugTools/DebugInterface.cpp b/pcsx2/DebugTools/DebugInterface.cpp
index d29c2b06f..afacbab20 100644
--- a/pcsx2/DebugTools/DebugInterface.cpp
+++ b/pcsx2/DebugTools/DebugInterface.cpp
@@ -20,12 +20,13 @@
 #include "R5900.h"
 #include "Debug.h"
 #include "VU.h"
-#include "GS.h" // Required for gsNonMirroredRead()
+#include "pcsx2/GS.h" // Required for gsNonMirroredRead()
 #include "Counters.h"
 
 #include "R3000A.h"
 #include "IopMem.h"
 #include "SymbolMap.h"
+#include "gui/AppCoreThread.h"
 
 R5900DebugInterface r5900Debug;
 R3000DebugInterface r3000Debug;
diff --git a/pcsx2/PAD/Linux/Config.h b/pcsx2/PAD/Linux/Config.h
index 26f46f0ed..39fae8d9b 100644
--- a/pcsx2/PAD/Linux/Config.h
+++ b/pcsx2/PAD/Linux/Config.h
@@ -12,10 +12,9 @@
  *  You should have received a copy of the GNU General Public License along with PCSX2.
  *  If not, see <http://www.gnu.org/licenses/>.
  */
-
 #pragma once
 
-#include "Global.h"
+#include "PAD/Linux/Global.h"
 
 class PADconf
 {
diff --git a/pcsx2/PAD/Linux/Global.h b/pcsx2/PAD/Linux/Global.h
index 255bb24c3..9bbeef22a 100644
--- a/pcsx2/PAD/Linux/Global.h
+++ b/pcsx2/PAD/Linux/Global.h
@@ -12,7 +12,6 @@
  *  You should have received a copy of the GNU General Public License along with PCSX2.
  *  If not, see <http://www.gnu.org/licenses/>.
  */
-
 #pragma once
 
 #include <stdio.h>
@@ -37,7 +36,7 @@
 static constexpr u32 GAMEPAD_NUMBER = 2;
 static constexpr u32 MAX_KEYS = 25;
 
-#include "Config.h"
+#include "PAD/Linux/Config.h"
 
 enum gamePadValues
 {
diff --git a/pcsx2/gui/App.h b/pcsx2/gui/App.h
index d9decd77a..b3650517c 100644
--- a/pcsx2/gui/App.h
+++ b/pcsx2/gui/App.h
@@ -36,7 +36,7 @@
 class DisassemblyDialog;
 struct HostKeyEvent;
 
-#include "GS.h"
+#include "pcsx2/GS.h"
 #include "System.h"
 #include "System/SysThreads.h"
 
diff --git a/pcsx2/gui/CpuUsageProvider.cpp b/pcsx2/gui/CpuUsageProvider.cpp
index d5db2b1db..223fb6e6a 100644
--- a/pcsx2/gui/CpuUsageProvider.cpp
+++ b/pcsx2/gui/CpuUsageProvider.cpp
@@ -22,8 +22,9 @@
 #include "System/SysThreads.h"
 #endif
 
-#include "GS.h"
+#include "pcsx2/GS.h"
 #include "MTVU.h"
+#include "AppCoreThread.h"
 
 void AllPCSX2Threads::LoadWithCurrentTimes()
 {
