--- libretro-parallel-n64-cfb9789e13ccbeaabd928203566346c05709d501/Makefile.orig	2020-10-27 21:37:22.337877895 +0100
+++ libretro-parallel-n64-cfb9789e13ccbeaabd928203566346c05709d501/Makefile	2020-10-27 21:37:52.921408086 +0100
@@ -236,7 +236,7 @@
       CXX = aarch64-linux-gnu-g++
       CPUFLAGS += -DNO_ASM -DARM -DARM_ASM -DDONT_WANT_ARM_OPTIMIZATIONS -DARM_FIX -DCLASSIC -DARM64
       LDFLAGS += -static-libgcc -static-libstdc++
-      GLES = 0
+      GLES = 1
       HAVE_NEON = 0
       WITH_DYNAREC=aarch64
       ifneq (,$(findstring neon,$(platform)))
