--- ./Makefile	2021-03-16 08:03:09.238353443 +0100
+++ ./Makefile	2021-03-16 09:10:28.314690450 +0100
@@ -172,18 +172,26 @@
       GLES = 1
       GL_LIB := -lGLESv2
       CPUFLAGS += -DNO_ASM -DARM -D__arm__ -DARM_ASM -D__NEON_OPT -DNOSSE -DARM_FIX
-      CPUFLAGS += -marm -mfloat-abi=hard
       HAVE_NEON = 1
       WITH_DYNAREC=arm
       ifneq (,$(findstring ODROIDC,$(BOARD)))
          # ODROID-C1
          CPUFLAGS += -mcpu=cortex-a5 -mfpu=neon
+         CPUFLAGS += -marm -mfloat-abi=hard
       else ifneq (,$(findstring ODROID-XU,$(BOARD)))
          # ODROID-XU3 XU4 and XU3-Lite
          CPUFLAGS += -mcpu=cortex-a15 -mtune=cortex-a15.cortex-a7 -mfpu=neon-vfpv4
+         CPUFLAGS += -marm -mfloat-abi=hard
+      else ifneq (,$(findstring ODROIDGO2,$(BOARD)))
+         # ODROIDGO2 and ODROIDGO3 (64 bits)
+         HAVE_PARALLEL = 1
+         WITH_DYNAREC=aarch64
+         CPUFLAGS += -DCLASSIC -DARM64
+         CPUFLAGS += -march=armv8-a+crc+simd -mtune=cortex-a35 -mcpu=cortex-a35
       else
          # ODROID-U3, U2, X2 & X
-          CPUFLAGS += -mcpu=cortex-a9 -mfpu=neon
+         CPUFLAGS += -mcpu=cortex-a9 -mfpu=neon
+         CPUFLAGS += -marm -mfloat-abi=hard
       endif
    endif
    
@@ -938,7 +946,7 @@
 endif
 
 # Fix for GCC 10, make sure its added to all stages of the compiler
-ifeq "$(shell expr `gcc -dumpversion` \>= 10)" "1"
+ifeq "$(shell expr `$(CC) -dumpversion` \>= 10)" "1"
   CPUFLAGS += -fcommon
 endif
 
