diff --git a/linux/linux.mk b/linux/linux.mk
index 701d7ee..24c4b8d 100644
--- a/linux/linux.mk
+++ b/linux/linux.mk
@@ -102,12 +102,23 @@ LINUX_POST_EXTRACT_HOOKS += LINUX_XTENSA_OVERLAY_EXTRACT
 LINUX_EXTRA_DOWNLOADS += $(ARCH_XTENSA_OVERLAY_URL)
 endif
 
+ifeq ($(BR2_LINUX_X64_KERNEL_IN_X32_USER),y)
+# make 64-bit kernel in 32-bit userspace
+LINUX_DEPENDENCIES += host-toolchain-optional-linaro-aarch64
+LINUX_MAKE_FLAGS = \
+	HOSTCC="$(HOSTCC) $(HOST_CFLAGS) $(HOST_LDFLAGS)" \
+	ARCH=arm64 \
+	INSTALL_MOD_PATH=$(TARGET_DIR) \
+	CROSS_COMPILE="$(HOST_DIR)/lib/gcc-linaro-aarch64-linux-gnu/bin/aarch64-linux-gnu-" \
+	DEPMOD=$(HOST_DIR)/sbin/depmod
+else
 LINUX_MAKE_FLAGS = \
 	HOSTCC="$(HOSTCC) $(HOST_CFLAGS) $(HOST_LDFLAGS)" \
 	ARCH=$(KERNEL_ARCH) \
 	INSTALL_MOD_PATH=$(TARGET_DIR) \
 	CROSS_COMPILE="$(TARGET_CROSS)" \
 	DEPMOD=$(HOST_DIR)/sbin/depmod
+endif
 
 LINUX_MAKE_ENV = \
 	$(TARGET_MAKE_ENV) \
@@ -190,6 +201,9 @@ ifeq ($(KERNEL_ARCH),i386)
 KERNEL_ARCH_PATH = $(LINUX_DIR)/arch/x86
 else ifeq ($(KERNEL_ARCH),x86_64)
 KERNEL_ARCH_PATH = $(LINUX_DIR)/arch/x86
+else ifeq ($(BR2_LINUX_X64_KERNEL_IN_X32_USER),y)
+# arm64 kernel when arch is arm
+KERNEL_ARCH_PATH = $(LINUX_DIR)/arch/arm64
 else
 KERNEL_ARCH_PATH = $(LINUX_DIR)/arch/$(KERNEL_ARCH)
 endif
