--- a/input/drivers/udev_input.c	2020-12-01 16:34:22.673375965 -0500
+++ b/input/drivers/udev_input.c	2020-12-07 18:32:39.295599000 -0500
@@ -803,7 +803,6 @@
 static int16_t udev_lightgun_aiming_state(
       udev_input_t *udev, unsigned port, unsigned id )
 {
-   struct video_viewport vp;
    const int edge_detect       = 32700;
    bool inside                 = false;
    int16_t res_x               = 0;
@@ -813,26 +812,11 @@
 
    udev_input_mouse_t *mouse   = udev_get_mouse(udev, port);
 
-   vp.x                        = 0;
-   vp.y                        = 0;
-   vp.width                    = 0;
-   vp.height                   = 0;
-   vp.full_width               = 0;
-   vp.full_height              = 0;
-
    if (!mouse)
       return 0;
 
-#ifdef HAVE_X11
-   /* udev->pointer_x and y is only set in X11 */
-   if (!(video_driver_translate_coord_viewport_wrap(
-               &vp, udev->pointer_x, udev->pointer_y,
-               &res_x, &res_y, &res_screen_x, &res_screen_y)))
-      return 0;
-#else
    res_x = udev_mouse_get_pointer_x(mouse, false);
    res_y = udev_mouse_get_pointer_y(mouse, false);
-#endif
 
    inside =    (res_x >= -edge_detect) 
             && (res_y >= -edge_detect)
@@ -842,11 +826,9 @@
    switch ( id )
    {
       case RETRO_DEVICE_ID_LIGHTGUN_SCREEN_X:
-         if (inside)
             return res_x;
          break;
       case RETRO_DEVICE_ID_LIGHTGUN_SCREEN_Y:
-         if (inside)
             return res_y;
          break;
       case RETRO_DEVICE_ID_LIGHTGUN_IS_OFFSCREEN:
@@ -1207,9 +1189,7 @@
    struct udev_list_entry     *devs = NULL;
    struct udev_list_entry     *item = NULL;
    struct udev_enumerate *enumerate = udev_enumerate_new(udev->udev);
-#ifdef DEBUG
    int device_index                 = 0;
-#endif
 
    if (!enumerate)
       return false;
@@ -1234,7 +1214,6 @@
          if (fd != -1)
          {
             bool check = udev_input_add_device(udev, type, devnode, cb);
-#ifdef DEBUG
             if (!check)
                RARCH_ERR("[udev] Failed to open device: %s (%s).\n",
                      devnode, strerror(errno));
@@ -1242,7 +1221,6 @@
                RARCH_LOG("[udev]: %s #%d (%s).\n",
                      type == UDEV_INPUT_KEYBOARD ? "Keyboard" : "Mouse",
                      device_index++, devnode);
-#endif
             (void)check;
             close(fd);
          }
