diff --git a/gfx/video_defines.h b/gfx/video_defines.h
index 38f8b58..7f2e84c 100644
--- a/gfx/video_defines.h
+++ b/gfx/video_defines.h
@@ -56,6 +56,7 @@ enum aspect_ratio
    ASPECT_RATIO_CORE,
    ASPECT_RATIO_CUSTOM,
    ASPECT_RATIO_FULL,
+   ASPECT_RATIO_CORE_CRT_1920,
 
    ASPECT_RATIO_END
 };

diff --git a/gfx/video_driver.c b/gfx/video_driver.c
index 69baa26..9b8a72a 100644
--- a/gfx/video_driver.c
+++ b/gfx/video_driver.c
@@ -2025,10 +2025,17 @@ void video_driver_set_viewport_core(void)
 
    /* Fallback to 1:1 pixel ratio if none provided */
    if (geom->aspect_ratio > 0.0f)
+   {
       aspectratio_lut[ASPECT_RATIO_CORE].value = geom->aspect_ratio;
+      aspectratio_lut[ASPECT_RATIO_CORE_CRT_1920].value = geom->aspect_ratio * 6;
+   }
    else
+   {
       aspectratio_lut[ASPECT_RATIO_CORE].value =
          (float)geom->base_width / geom->base_height;
+      aspectratio_lut[ASPECT_RATIO_CORE_CRT_1920].value =
+         6 * (float)geom->base_width / geom->base_height;
+   }
 }
 
 void video_driver_set_rgba(void)
