diff --git a/network/netplay/netplay_frontend.c b/network/netplay/netplay_frontend.c
index 2959de4..83f3749 100644
--- a/network/netplay/netplay_frontend.c
+++ b/network/netplay/netplay_frontend.c
@@ -7168,7 +7168,7 @@ static void netplay_announce(void)
             frontend_architecture_tmp, sizeof(frontend_architecture_tmp));
    snprintf(frontend_architecture, 
          sizeof(frontend_architecture),
-         "%s %s",
+         "%s %s@RECALBOX",
          frontend_drv->ident,
          frontend_architecture_tmp);
