diff --git a/input/drivers/udev_input.c b/input/drivers/udev_input.c
index 979425f..c85f7a3 100644
--- a/input/drivers/udev_input.c
+++ b/input/drivers/udev_input.c
@@ -847,9 +847,11 @@ static int16_t udev_lightgun_aiming_state(
    switch ( id )
    {
       case RETRO_DEVICE_ID_LIGHTGUN_SCREEN_X:
+         if (inside)
             return res_x;
          break;
       case RETRO_DEVICE_ID_LIGHTGUN_SCREEN_Y:
+         if (inside)
             return res_y;
          break;
       case RETRO_DEVICE_ID_LIGHTGUN_IS_OFFSCREEN:
(END)
