diff --git a/retroarch.c b/retroarch.c
index 6a88c3108e..d4a33415cf 100644
--- a/retroarch.c
+++ b/retroarch.c
@@ -4727,7 +4727,7 @@ int generic_menu_entry_action(
                   entry->label, entry->type, i, entry->entry_idx);
          break;
       case MENU_ACTION_SEARCH:
-         menu_input_dialog_start_search();
+         //menu_input_dialog_start_search();
          break;
       case MENU_ACTION_SCAN:
          if (cbs && cbs->action_scan)
