diff --git a/input/drivers/udev_input.c b/input/drivers/udev_input.c
index c6bec5e3ab..2b39a98e06 100644
--- a/input/drivers/udev_input.c
+++ b/input/drivers/udev_input.c
@@ -807,9 +807,6 @@ static bool udev_pointer_is_off_window(const udev_input_t *udev)
 static int16_t udev_lightgun_aiming_state(
       udev_input_t *udev, unsigned port, unsigned id )
 {
-#ifdef HAVE_X11
-   struct video_viewport vp;
-#endif
    const int edge_detect       = 32700;
    bool inside                 = false;
    int16_t res_x               = 0;
@@ -819,28 +816,11 @@ static int16_t udev_lightgun_aiming_state(
 
    udev_input_mouse_t *mouse   = udev_get_mouse(udev, port);
 
-#ifdef HAVE_X11
-   vp.x                        = 0;
-   vp.y                        = 0;
-   vp.width                    = 0;
-   vp.height                   = 0;
-   vp.full_width               = 0;
-   vp.full_height              = 0;
-#endif
-
    if (!mouse)
       return 0;
 
-#ifdef HAVE_X11
-   /* udev->pointer_x and y is only set in X11 */
-   if (!(video_driver_translate_coord_viewport_wrap(
-               &vp, udev->pointer_x, udev->pointer_y,
-               &res_x, &res_y, &res_screen_x, &res_screen_y)))
-      return 0;
-#else
-   res_x  = udev_mouse_get_pointer_x(mouse, false);
-   res_y  = udev_mouse_get_pointer_y(mouse, false);
-#endif
+   res_x = udev_mouse_get_pointer_x(mouse, false);
+   res_y = udev_mouse_get_pointer_y(mouse, false);
 
    inside =    (res_x >= -edge_detect) 
             && (res_y >= -edge_detect)
@@ -850,11 +830,9 @@ static int16_t udev_lightgun_aiming_state(
    switch ( id )
    {
       case RETRO_DEVICE_ID_LIGHTGUN_SCREEN_X:
-         if (inside)
             return res_x;
          break;
       case RETRO_DEVICE_ID_LIGHTGUN_SCREEN_Y:
-         if (inside)
             return res_y;
          break;
       case RETRO_DEVICE_ID_LIGHTGUN_IS_OFFSCREEN:
@@ -1227,9 +1205,7 @@ static bool open_devices(udev_input_t *udev,
    struct udev_list_entry     *devs = NULL;
    struct udev_list_entry     *item = NULL;
    struct udev_enumerate *enumerate = udev_enumerate_new(udev->udev);
-#ifdef DEBUG
    int device_index                 = 0;
-#endif
 
    if (!enumerate)
       return false;
@@ -1254,7 +1230,6 @@ static bool open_devices(udev_input_t *udev,
          if (fd != -1)
          {
             bool check = udev_input_add_device(udev, type, devnode, cb);
-#ifdef DEBUG
             if (!check)
                RARCH_ERR("[udev] Failed to open device: %s (%s).\n",
                      devnode, strerror(errno));
@@ -1262,7 +1237,6 @@ static bool open_devices(udev_input_t *udev,
                RARCH_LOG("[udev]: %s #%d (%s).\n",
                      type == UDEV_INPUT_KEYBOARD ? "Keyboard" : "Mouse",
                      device_index++, devnode);
-#endif
             (void)check;
             close(fd);
          }
