--- a/mosquitto.conf	2020-02-28 00:49:51.000000000 +0100
+++ b/mosquitto.conf	2020-12-04 20:16:38.705699354 +0100
@@ -208,7 +208,7 @@
 #bind_address
 
 # Port to use for the default listener.
-#port 1883
+port 1883
 
 # Bind the listener to a specific interface. This is similar to
 # bind_address above but is useful when an interface has multiple addresses or
@@ -237,7 +237,7 @@
 # Websockets support is currently disabled by default at compile time.
 # Certificate based TLS may be used with websockets, except that
 # only the cafile, certfile, keyfile and ciphers options are supported.
-#protocol mqtt
+protocol mqtt
 
 # Set use_username_as_clientid to true to replace the clientid that a client
 # connected with with its username. This allows authentication to be tied to
@@ -371,7 +371,7 @@
 # Note that for a websockets listener it is not possible to bind to a host
 # name.
 # listener port-number [ip address/host name]
-#listener
+listener 18833
 
 # Bind the listener to a specific interface. This is similar to
 # the [ip address/host name] part of the listener definition, but is useful
@@ -408,7 +408,7 @@
 # This can be either mqtt or websockets.
 # Certificate based TLS may be used with websockets, except that only the
 # cafile, certfile, keyfile and ciphers options are supported.
-#protocol mqtt
+protocol websockets
 
 # Set use_username_as_clientid to true to replace the clientid that a client
 # connected with with its username. This allows authentication to be tied to
