--- ./mosquitto.conf.orig	2021-06-21 14:29:08.312363585 +0200
+++ ./mosquitto.conf	2021-06-21 16:06:09.175479059 +0200
@@ -871,4 +871,4 @@
 # alphabetical order, with capital letters ordered first. If this option is
 # given multiple times, all of the files from the first instance will be
 # processed before the next instance. See the man page for examples.
-#include_dir
+include_dir /etc/mosquitto.d
