diff -u switchres-a8af8e8f7f54d24414b4a8595a84c0f0f23e14a1/display.cpp switchres-modified/display.cpp
--- a/display.cpp	2021-06-28 23:51:06.000000000 +0200
+++ b/display.cpp	2021-08-10 23:03:21.249614251 +0200
@@ -468,6 +480,7 @@
 	// Create a working range with the best possible information
 	if (desktop_mode.type & CUSTOM_VIDEO_TIMING_SYSTEM) modeline_vesa_gtf(&desktop_mode);
 	modeline_to_monitor_range(range, &desktop_mode);
+	range->progressive_lines_min = 480;
 	monitor_show_range(range);
 
 	// Force our resolution to LCD's native one

--- a/display.cpp	2021-08-11 00:12:06.930494694 +0200
+++ b/display.cpp	2021-08-11 00:22:40.844767991 +0200
@@ -300,16 +300,7 @@
 		if (mode.type & MODE_DESKTOP)
 			mode.type &= ~MODE_DISABLED;
 
-		// Lock all modes that don't match the user's -resolution rules
-		if (m_user_mode.width != 0 || m_user_mode.height != 0 || m_user_mode.refresh == !0)
-		{
-			if (!( (mode.width == m_user_mode.width || (mode.type & X_RES_EDITABLE) || m_user_mode.width == 0)
-				&& (mode.height == m_user_mode.height || (mode.type & Y_RES_EDITABLE) || m_user_mode.height == 0)
-				&& (mode.refresh == m_user_mode.refresh || (mode.type & V_FREQ_EDITABLE) || m_user_mode.refresh == 0) ))
-				mode.type |= MODE_DISABLED;
-			else
-				mode.type &= ~MODE_DISABLED;
-		}
+		mode.type &= ~MODE_DISABLED;
 	}
 
 	return true;
