--- a/Core/FirmwareHelper.h	2021-11-25 20:28:28.000000000 +0100
+++ b/Core/FirmwareHelper.h	2021-11-25 20:31:03.000000000 +0100
@@ -100,7 +100,7 @@
 		}
 
 		MissingFirmwareMessage msg;
-		msg.Filename = "BS-X.bin";
+		msg.Filename = "satellaview/BS-X.bin";
 		msg.Firmware = FirmwareType::Satellaview;
 		msg.Size = 1024*1024;
 		console->GetNotificationManager()->SendNotification(ConsoleNotificationType::MissingFirmware, &msg);
@@ -141,10 +141,10 @@
 		string altFilename;
 		switch(type) {
 			default:
-			case FirmwareType::Gameboy: filename = "dmg_boot.bin"; altFilename = "gb_bios.bin"; break;
-			case FirmwareType::GameboyColor: filename = "cgb_boot.bin"; altFilename = "gbc_bios.bin"; break;
-			case FirmwareType::Sgb1GameboyCpu: filename = "sgb_boot.bin"; altFilename = "sgb_bios.bin"; break;
-			case FirmwareType::Sgb2GameboyCpu: filename = "sgb2_boot.bin"; altFilename = "sgb_bios.bin"; break;
+			case FirmwareType::Gameboy: filename = "gb/dmg_boot.bin"; altFilename = "gb_bios.bin"; break;
+			case FirmwareType::GameboyColor: filename = "gbc/cgb_boot.bin"; altFilename = "gbc_bios.bin"; break;
+			case FirmwareType::Sgb1GameboyCpu: filename = "sgb_boot.bin"; altFilename = "sgb/sgb_bios.bin"; break;
+			case FirmwareType::Sgb2GameboyCpu: filename = "sgb/sgb2_boot.bin"; altFilename = "sgb_bios.bin"; break;
 		}
 
 		uint32_t size = type == FirmwareType::GameboyColor ? 2304 : 256;
