From 2c2f056954cac28de96d8fb4dda3d59d82ef7cc5 Mon Sep 17 00:00:00 2001
From: Chips-fr <darcelf@gmail.com>
Date: Mon, 19 Jul 2021 18:59:15 +0200
Subject: [PATCH] make kickstart names as recalbox

---
 libretro/core/libretro-core.cpp | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/libretro/core/libretro-core.cpp b/libretro/core/libretro-core.cpp
index 459f57e..2137003 100644
--- a/libretro/core/libretro-core.cpp
+++ b/libretro/core/libretro-core.cpp
@@ -67,12 +67,12 @@ static size_t save_state_file_size = 0;
 
 // Amiga default kickstarts
 
-#define A500_ROM        "kick34005.A500"
-#define A600_ROM        "kick40063.A600"
-#define A1200_ROM       "kick40068.A1200"
-#define CDTV_ROM        "kick34005.CDTV"
-#define CD32_ROM        "kick40060.CD32"
-#define CD32_ROM_EXT    "kick40060.CD32.ext"
+#define A500_ROM        "kick34005.A500.rom"
+#define A600_ROM        "kick40063.A600.rom"
+#define A1200_ROM       "kick40068.A1200.rom"
+#define CDTV_ROM        "kick34005.CDTV.rom"
+#define CD32_ROM        "kick40060.CD32.rom"
+#define CD32_ROM_EXT    "kick40060.CD32.ext.rom"
 
 bool isCD32 = false;
 
-- 
2.25.1

