diff --git a/src/sdl/ti994a-sdl.cpp b/src/sdl/ti994a-sdl.cpp
index 934f02b..924010c 100644
--- a/src/sdl/ti994a-sdl.cpp
+++ b/src/sdl/ti994a-sdl.cpp
@@ -245,11 +245,11 @@ void cSdlTI994A::Run( const std::string& imgFile )
 				if( joystick != -1 )
 				{
 					pic->SetJoystickButton( joystick, true );
-					if( event.jbutton.button > 0 )
+					/*if( event.jbutton.button > 0 )
 					{
 						auto vkey = ( VIRTUAL_KEY_E ) ( VK_0 + std::clamp(( int ) event.jbutton.button, 1, 9 ));
 						pic->VKeysDown( 0, vkey );
-					}
+					}*/
 				}
 				break;
 			case SDL_JOYBUTTONUP :
@@ -257,10 +257,10 @@ void cSdlTI994A::Run( const std::string& imgFile )
 				if( joystick != -1 )
 				{
 					pic->SetJoystickButton( joystick, false );
-					if( event.jbutton.button > 0 )
+					/*if( event.jbutton.button > 0 )
 					{
 						pic->VKeyUp( 0 );
-					}
+					}*/
 				}
 				break;
 			case SDL_KEYDOWN :
