diff --git a/include/icomputer.hpp b/include/icomputer.hpp
index 905afcb..5fd1903 100644
--- a/include/icomputer.hpp
+++ b/include/icomputer.hpp
@@ -68,7 +68,7 @@ struct iComputer :
 	virtual bool Sleep( int, UINT32 ) = 0;
 	virtual bool WakeCPU( UINT32 ) = 0;
 
-	virtual void Run( ) = 0;
+	virtual void Run( const std::string& imgFile ) = 0;
 	virtual void Stop( ) = 0;
 	virtual bool Step( ) = 0;
 	virtual bool IsRunning( ) = 0;
diff --git a/include/ti994a-console.hpp b/include/ti994a-console.hpp
index 887b417..b050ebd 100644
--- a/include/ti994a-console.hpp
+++ b/include/ti994a-console.hpp
@@ -53,7 +53,7 @@ public:
 	// iComputer methods
 	virtual int ReadCRU( ADDRESS ) override;
 	virtual void WriteCRU( ADDRESS, UINT16 ) override;
-	virtual void Run( ) override;
+	virtual void Run( const std::string& imgFile ) override;
 	virtual bool Step( ) override;
 	virtual bool LoadImage( const char * ) override;
 
diff --git a/include/ti994a-sdl.hpp b/include/ti994a-sdl.hpp
index 91a3b6c..44107e3 100644
--- a/include/ti994a-sdl.hpp
+++ b/include/ti994a-sdl.hpp
@@ -61,7 +61,7 @@ public:
 	virtual void Reset( ) override;
 	virtual bool Sleep( int, UINT32 ) override;
 	virtual bool WakeCPU( UINT32 ) override;
-	virtual void Run( ) override;
+	virtual void Run( const std::string& imgFile ) override;
 	virtual bool SaveImage( const char * ) override;
 	virtual bool LoadImage( const char * ) override;
 
diff --git a/include/ti994a.hpp b/include/ti994a.hpp
index 304684f..960d839 100644
--- a/include/ti994a.hpp
+++ b/include/ti994a.hpp
@@ -119,7 +119,7 @@ public:
 	virtual void Reset( ) override;
 	virtual bool Sleep( int, UINT32 ) override;
 	virtual bool WakeCPU( UINT32 ) override;
-	virtual void Run( ) override;
+	virtual void Run( const std::string& imgFile ) override;
 	virtual void Stop( ) override;
 	virtual bool Step( ) override;
 	virtual bool IsRunning( ) override;
diff --git a/src/console/ti-main.cpp b/src/console/ti-main.cpp
index 83226b8..67865f8 100644
--- a/src/console/ti-main.cpp
+++ b/src/console/ti-main.cpp
@@ -224,16 +224,10 @@ int main( int argc, char *argv[] )
 		computer->InsertCartridge( ctg );
 	}
 
-	if( !imgFile.empty( ))
-	{
-		if( verbose >= 1 )
-		{
-			fprintf( stdout, "Loading image \"%s\"\n", imgFile.c_str( ));
-		}
-		computer->LoadImage( imgFile.c_str( ));
-	}
+	if( imgFile.empty( ))
+	  imgFile = "/recalbox/share/saves/ti994a/snapshots/" + std::filesystem::path(ctgFile).stem().string() + ".img";
 
-	computer->Run( );
+	computer->Run( imgFile );
 
 	ClearScreen( );
 	ShowCursor( );
diff --git a/src/console/ti994a-console.cpp b/src/console/ti994a-console.cpp
index fd35051..076018b 100644
--- a/src/console/ti994a-console.cpp
+++ b/src/console/ti994a-console.cpp
@@ -186,7 +186,7 @@ void cConsoleTI994A::WriteCRU( ADDRESS address, UINT16 data )
 	}
 }
 
-void cConsoleTI994A::Run( )
+void cConsoleTI994A::Run( const std::string& imgFile )
 {
 	m_CPU->RegisterDebugHandler( _DebugHandler, this );
 
diff --git a/src/core/ti994a-gk.cpp b/src/core/ti994a-gk.cpp
index 101a7f8..15771bf 100644
--- a/src/core/ti994a-gk.cpp
+++ b/src/core/ti994a-gk.cpp
@@ -134,7 +134,7 @@ std::optional<sStateSection> cTI994AGK::SaveState( )
 
 	auto save = cTI994A::SaveState( );
 
-	if( save )
+	if( save && m_GK_Cartridge )
 	{
 		save->store( "GK.ROM", m_GK_Cartridge->GetDescriptor( ));
 		save->addSubSection( m_GK_Cartridge );
diff --git a/src/core/ti994a.cpp b/src/core/ti994a.cpp
index 429021c..39907bb 100644
--- a/src/core/ti994a.cpp
+++ b/src/core/ti994a.cpp
@@ -590,7 +590,7 @@ void cTI994A::WriteCRU( ADDRESS address, UINT16 val )
 	}
 }
 
-void cTI994A::Run( )
+void cTI994A::Run( const std::string& imgFile )
 {
 	FUNCTION_ENTRY( this, "cTI994A::Run", true );
 
@@ -624,9 +624,7 @@ bool cTI994A::SaveImage( const char *filename )
 
 	if( auto save = SaveState( ))
 	{
-		CreateHomePath( );
-
-		auto path = std::filesystem::path{ GetHomePath( )} / std::filesystem::path( filename );
+		auto path = std::filesystem::path(filename);
 
 		save->SaveImage( path );
 
@@ -644,7 +642,7 @@ bool cTI994A::LoadImage( const char *filename )
 
 	try
 	{
-		auto path = std::filesystem::path{ GetHomePath( )} / std::filesystem::path( filename );
+		auto path = std::filesystem::path(filename);
 
 		if( auto save =  sStateSection::LoadImage( path ))
 		{
diff --git a/src/sdl/main.cpp b/src/sdl/main.cpp
index 08bc10c..5e88257 100644
--- a/src/sdl/main.cpp
+++ b/src/sdl/main.cpp
@@ -512,14 +512,8 @@ int main( int argc, char *argv[] )
 		computer->InsertCartridge( ctg );
 	}
 
-	if( !imgFile.empty( ))
-	{
-		if( verbose >= 1 )
-		{
-			fprintf( stdout, "Loading image \"%s\"\n", imgFile.c_str( ));
-		}
-		computer->LoadImage( imgFile.c_str( ));
-	}
+	if( imgFile.empty( ))
+	  imgFile = "/recalbox/share/saves/ti994a/snapshots/" + std::filesystem::path(ctgFile).stem().string() + ".img";
 
 	if( verbose >= 1 )
 	{
@@ -530,7 +524,7 @@ int main( int argc, char *argv[] )
 		}
 	}
 
-	computer->Run( );
+	computer->Run( imgFile );
 
 	if( joy1 != nullptr )
 	{
diff --git a/src/sdl/ti994a-sdl.cpp b/src/sdl/ti994a-sdl.cpp
index 3c0da4a..934f02b 100644
--- a/src/sdl/ti994a-sdl.cpp
+++ b/src/sdl/ti994a-sdl.cpp
@@ -40,7 +40,7 @@
 
 DBG_REGISTER( __FILE__ );
 
-const char SAVE_IMAGE[ ] = "ti-994a.img";
+const char SAVE_IMAGE[ ] = "/recalbox/share/saves/ti994a/snapshots/snapshot.img";
 
 extern int verbose;
 
@@ -160,7 +160,7 @@ void cSdlTI994A::Reset( )
 	}
 }
 
-void cSdlTI994A::Run( )
+void cSdlTI994A::Run( const std::string& imgFile )
 {
 	FUNCTION_ENTRY( this, "cSdlTI994A::Run", true );
 
@@ -314,10 +314,10 @@ void cSdlTI994A::Run( )
 					switch( event.key.keysym.sym )
 					{
 						case SDLK_F2 :
-							SaveImage( SAVE_IMAGE );
+							SaveImage( imgFile.data() );
 							break;
 						case SDLK_F3 :
-							LoadImage( SAVE_IMAGE );
+						  LoadImage( imgFile.data() );
 							break;
 						case SDLK_F10 :
 							Reset( );
