--- a/libretro/libretro.c	2022-07-06 20:58:49.000000000 +0200
+++ b/libretro/libretro.c	2022-07-07 10:45:25.000000000 +0200
@@ -3155,14 +3155,14 @@
    snprintf(AR_ROM, sizeof(AR_ROM), "%s%careplay.bin", dir, slash);
    snprintf(SK_ROM, sizeof(SK_ROM), "%s%csk.bin", dir, slash);
    snprintf(SK_UPMEM, sizeof(SK_UPMEM), "%s%csk2chip.bin", dir, slash);
-   snprintf(MD_BIOS, sizeof(MD_BIOS), "%s%cbios_MD.bin", dir, slash);
-   snprintf(GG_BIOS, sizeof(GG_BIOS), "%s%cbios.gg", dir, slash);
-   snprintf(MS_BIOS_EU, sizeof(MS_BIOS_EU), "%s%cbios_E.sms", dir, slash);
-   snprintf(MS_BIOS_US, sizeof(MS_BIOS_US), "%s%cbios_U.sms", dir, slash);
-   snprintf(MS_BIOS_JP, sizeof(MS_BIOS_JP), "%s%cbios_J.sms", dir, slash);
-   snprintf(CD_BIOS_EU, sizeof(CD_BIOS_EU), "%s%cbios_CD_E.bin", dir, slash);
-   snprintf(CD_BIOS_US, sizeof(CD_BIOS_US), "%s%cbios_CD_U.bin", dir, slash);
-   snprintf(CD_BIOS_JP, sizeof(CD_BIOS_JP), "%s%cbios_CD_J.bin", dir, slash);
+   snprintf(MD_BIOS, sizeof(MD_BIOS), "%s%c%s%cbios_MD.bin", dir, slash, "megadrive", slash);
+   snprintf(GG_BIOS, sizeof(GG_BIOS), "%s%c%s%cbios.gg", dir, slash, "gamegear", slash);
+   snprintf(MS_BIOS_EU, sizeof(MS_BIOS_EU), "%s%c%s%cbios_E.sms", dir, slash, "mastersystem", slash);
+   snprintf(MS_BIOS_US, sizeof(MS_BIOS_US), "%s%c%s%cbios_U.sms", dir, slash, "mastersystem", slash);
+   snprintf(MS_BIOS_JP, sizeof(MS_BIOS_JP), "%s%c%s%cbios_J.sms", dir, slash, "mastersystem", slash);
+   snprintf(CD_BIOS_EU, sizeof(CD_BIOS_EU), "%s%c%s%cbios_CD_E.bin", dir, slash, "segacd", slash);
+   snprintf(CD_BIOS_US, sizeof(CD_BIOS_US), "%s%c%s%cbios_CD_U.bin", dir, slash, "segacd", slash);
+   snprintf(CD_BIOS_JP, sizeof(CD_BIOS_JP), "%s%c%s%cbios_CD_J.bin", dir, slash, "segacd", slash);
    snprintf(CART_BRAM, sizeof(CART_BRAM), "%s%ccart.brm", save_dir, slash);

    check_variables(true);
